lib/src/back_end/code_writer.dart

```diff
@@ -134,7 +131,10 @@ class CodeWriter {
   ///
   // TODO(tall): Add another API that adds/subtracts existing indentation.
   void setIndent(int indent) {
-    _options.indent = _pieceOptions[_pieceOptions.length - 2].indent + indent;
+    var parentIndent =
+        _options.length > 1 ? _options[_options.length - 2].indent : 0;
+
+    _options.last.indent = parentIndent + indent;
```

Comment:
Nit: I know you're just renaming this, but this reminded me how hard it was to read these couple lines to figure out how the indentation worked. Could we add a comment, an example or something to make this a little more readable?

Reply:
Done. I don't really like how the options stack works either. I thought about reworking it here, but I felt like there was enough change already.